### PR TITLE
Add ability to search from Office

### PR DESCRIPTION
**Pull request by andvirga**
Jira: https://bugs.benetech.org/browse/CHI-529

Related to HRM changes in: https://github.com/tech-matters/hrm/pull/101

Primary reviewer: @GPaoloni 

Changes description:
- Added a new dropdown to filter Contact and Cases by Offices (this dropdown won't appear if the user has a helpline attribute set or if `OfficeInformation` is not available).
- Moved `OfficeInformation` to Form Definitions.
- Defined `OfficeDefinitions` and `OfficeEntry` types.
- Created a helper to get Office data from `OfficeInformation`.
- Added a new variable (multipleOfficeSupport) that enables the Office display and search dropdown. (To be moved to service config).

### CODE REVIEW

**Review by GPaoloni — APPROVED** [review]
Looks and works great to me!
Just one thing is that the "all cases list" is not filtering by helpline when the counselor has one, but you can easily solve that sending `&helpline=${helpline}` as part of the query parameters for the /cases endpoint.

**Comment by andvirga** [comment]
> Looks and works great to me!
> Just one thing is that the "all cases list" is not filtering by helpline when the counselor has one, but you can easily solve that sending `&helpline=${helpline}` as part of the query parameters for the /cases endpoint.

Good catch! Thanks!

File: plugin-hrm-form/src/components/common/forms/types.ts [hunk 1]
```diff
@@ -120,6 +120,17 @@ type CallTypeButtonsEntry = {
...
+export type OfficeEntry = {
+  name: string;
+  manager?: {
+    name: string;
+    phone: string;
+    email: string;
+  };
+};
+
+export type OfficeDefinitions = OfficeEntry[];
```
[the remark this answers is not in the document]
Comment (andvirga) [reply plugin-hrm-form/src/components/common/forms/types.ts:132]: I'll leave a comment to improve this in a future story! Thanks!

File: plugin-hrm-form/src/components/search/SearchForm/index.jsx [hunk 2]
```diff
@@ -32,30 +32,54 @@ class SearchForm extends Component {
...
+    officeInformation: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string,
+        manager: PropTypes.shape({
+          name: string,
+          phone: string,
+          email: string,
+        }),
+      }),
+    ).isRequired,
```
[the remark this answers is not in the document]
Comment (nickhurlburt) [reply plugin-hrm-form/src/components/search/SearchForm/index.jsx]: +1!  Keeping the console clear of noise is worth it.

File: plugin-hrm-form/src/components/search/SearchForm/index.jsx [hunk 3]
```diff
@@ -122,7 +146,20 @@ class SearchForm extends Component {
...
               onKeyPress={submitOnEnter}
+              style={{ marginRight: 25 }}
             />
+            {/* If the user has their helpline attribute set, we don't need to show the Office search criteria. */}
+            {multipleOfficeSupport && !userHelpline && officeOptions.length > 0 && (
+              <FieldSelect
+                id="Search_Office"
+                name="office"
+                label={strings['SearchForm-Office']}
+                placeholder="--"
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/components/search/SearchForm/index.jsx:157]: When I select this, I'm getting this warning in the console:
```
plugin-hrm-form.js:367616 Warning: A component is changing an uncontrolled input of type hidden to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (created by SelectInput)
    in div (created by SelectInput)
    in SelectInput (created by InputBase)
    in div (created by InputBase)
    in InputBase (created by Context.Consumer)
    in WithFormControlContext(InputBase) (created by t)
    in t (created by Input)
    in Input (created by t)
    in t
    in Select (created by Context.Consumer)
    in WithFormControlContext(Select) (created by t)
    in t (at HrmStyles.tsx:188)
```

**Review by nickhurlburt — APPROVED** [review]
Looks good, with a few comments

**Pull request merged**